src/nlinv.c

```diff
@@ -145,9 +142,13 @@ int main_nlinv(int argc, char* argv[])
 
 		md_zsmul(DIMS, trj_dims, traj, traj, 2.);
 
-		//if (0 == md_calc_size(3, sens_dims))
-			estimate_fast_sq_im_dims(3, dims, trj_dims, traj);
+		estimate_im_dims(DIMS, FFT_FLAGS, dims, trj_dims, traj);
+		debug_printf(DP_INFO, "Est. image size: %ld %ld %ld\n", dims[0], dims[1], dims[2]);
+		md_copy_dims(DIMS - 3, dims + 3, ksp_dims + 3);
+	} else {
+		md_copy_dims(DIMS, dims, ksp_dims);
 	}
+	dims[MAPS_DIM] = nmaps;
```

Comment:
I prefer this without the else clause as it was before, simply replace the call to estimate_fast_sq_dims with estimate_im_dims.

Reply:
Hm, nufft also works with twice the grid size internally... we should make it fully consistent if possible.. Maybe nufft does not scale the trajectory but then doubles the grid size?  This may be cleaner and could avoid odd grid sizes.

Reply:
> I prefer this without the else clause as it was before, simply replace the call to estimate_fast_sq_dims with estimate_im_dims.

I have added another commit to avoid the else. The additional ```md_copy_dims(DIMS - 3, dims + 3, ksp_dims + 3);``` after ```estimate_im_dims()``` is however also necessary within the if clause because estimate_im_dims will reset dimensions not present in the trajectory.

Reply:
> Hm, nufft also works with twice the grid size internally... we should make it fully consistent if possible.. Maybe nufft does not scale the trajectory but then doubles the grid size? This may be cleaner and could avoid odd grid sizes.

I will take a look and see if I can find a solution (and then post it in a new merge request).